File: 프로그래머스/1/42840. 모의고사/모의고사.cs
using System;
using System.Collections.Generic;

public class Solution {
    public int[] solution(int[] answers) {
        int[] supo1 = new int[5] {1, 2, 3, 4, 5};
        int[] supo2 = new int[8] {2, 1, 2, 3, 2, 4, 2, 5};
        int[] supo3 = new int[10] {3, 3, 1, 1, 2, 2, 4, 4, 5, 5};
        int[] score = new int[3] {0, 0, 0};
        int max = 0;
        List<int> nList = new List<int>();
        
        for(int i = 0; i < answers.Length; i++)
        {
            if(supo1[i % 5] == answers[i]) { score[0]++; }
            if(supo2[i % 8] == answers[i]) { score[1]++; }
            if(supo3[i % 10] == answers[i]) { score[2]++; }
        }
        
        for(int i = 0; i < 3; i++) { if(max < score[i]) { max= score[i]; } }
        for(int i = 0; i < 3; i++) { if(max == score[i]) { nList.Add(i+1); } }
        
        return nList.ToArray();
    }
}

File: 프로그래머스/1/42840. 모의고사/모의고사.cpp
#include <string>
#include <vector>

using namespace std;

vector<int> solution(vector<int> answers) {
    int a = 0;
    int b = 0;
    int c = 0;
    vector<int>one = {1,2,3,4,5};
    vector<int>two = {2,1,2,3,2,4,2,5};
    vector<int>three = {3,3,1,1,2,2,4,4,5,5};
    vector<int> answer;
    
    for(int i = 0; i < answers.size(); i ++)
    {
        if(answers[i] == one[i%one.size()])
        {
            a++;
        }
        if(answers[i] == two[i%two.size()])
        {
            b++;
        }
        if(answers[i] == three[i%three.size()])
        {
            c++;
        }
        
        
    }
    if(a == b && b == c)
    {
        answer = {1,2,3};
    }
    else if(a == b && b > c)
    {
        answer = {1,2};
    }
    else if(a == c && c > b)
    {
        answer = {1,3};
    }
    else if(b == c && c > a)
    {
        answer = {2,3};
    }
    else if(a > c && a >b)
    {
        answer = {1};
    }
    else if(b > a && b > c){
        answer = {2};
    }
    else
    {
        answer = {3};
    }
    
    return answer;
}